guides/keras_cv/simsiam-with-kerascv.py

```diff
@@ -0,0 +1,638 @@
+    lambda image: tf.cast(image, tf.float32), num_parallel_calls=tf.data.AUTOTUNE
+)
+train_ds = train_ds.shuffle(buffer_size=8 * BATCH_SIZE, reshuffle_each_iteration=True)
+
+"""
+Next, we need to prepare some labelled samples.
+This is done so that TensorFlow similarity can probe the learned embedding to ensure
+that the model is learning appropriately.
+"""
+
+(x_raw_train, y_raw_train), ds_info = tfds.load(
+    "stl10", split="train", as_supervised=True, batch_size=-1, with_info=True
+)
+x_raw_train, y_raw_train = tf.cast(x_raw_train, tf.float32), tf.cast(
+    y_raw_train, tf.float32
+)
+x_test, y_test = tfds.load(
+    "stl10",
+    split="test",
+    as_supervised=True,
+    batch_size=-1,
+)
+x_test, y_test = tf.cast(x_test, tf.float32), tf.cast(y_test, tf.float32)
+
+"""
+In self supervised learning, queries and indexes are labeled subset datasets used to
+evaluate the quality of the produced latent embedding.  The following code assembles
+these datasets:
+"""
+
+# Compute the indicies for query, index, val, and train splits
+query_idxs, index_idxs, val_idxs, train_idxs = [], [], [], []
+for cid in range(ds_info.features["label"].num_classes):
+    idxs = tf.random.shuffle(tf.where(y_raw_train == cid))
+    idxs = tf.reshape(idxs, (-1,))
+    query_idxs.extend(idxs[:100])  # 200 query examples per class
+    index_idxs.extend(idxs[100:200])  # 200 index examples per class
+    val_idxs.extend(idxs[200:300])  # 100 validation examples per class
+    train_idxs.extend(idxs[300:])  # The remaining are used for training
+
+random.shuffle(query_idxs)
+random.shuffle(index_idxs)
+random.shuffle(val_idxs)
+random.shuffle(train_idxs)
+
+
+def create_split(idxs: list) -> tuple:
+    x, y = [], []
+    for idx in idxs:
+        x.append(x_raw_train[int(idx)])
+        y.append(y_raw_train[int(idx)])
+    return tf.convert_to_tensor(np.array(x), dtype=tf.float32), tf.convert_to_tensor(
+        np.array(y), dtype=tf.int64
+    )
+
+
+x_query, y_query = create_split(query_idxs)
+x_index, y_index = create_split(index_idxs)
+x_val, y_val = create_split(val_idxs)
+x_train, y_train = create_split(train_idxs)
+
+PRE_TRAIN_STEPS_PER_EPOCH = tf.data.experimental.cardinality(train_ds) // BATCH_SIZE
+PRE_TRAIN_STEPS_PER_EPOCH = int(PRE_TRAIN_STEPS_PER_EPOCH.numpy())
+
+print(
+    tabulate(
+        [
+            ["train", tf.data.experimental.cardinality(train_ds), None],
+            ["val", x_val.shape, y_val.shape],
+            ["query", x_query.shape, y_query.shape],
+            ["index", x_index.shape, y_index.shape],
+            ["test", x_test.shape, y_test.shape],
+        ],
+        headers=["# of Examples", "Labels"],
+    )
+)
+
+"""
+## Augmentations
+Self-supervised networks require at least two augmented "views" of each example. This can be created using a DataSet and an augmentation function. The DataSet treats each example in the batch as its own class and then the augment function produces two separate views for each example.
```

Comment:
Style nit: please keep text lines short, it helps with version control and reviews